src/core/qgsfeature.cpp

```diff
@@ -127,6 +127,23 @@ QgsAttributes QgsFeature::attributes() const
   return d->attributes;
 }
 
+QVariantMap QgsFeature::attributeMap() const
+{
+  QVariantMap res;
+  const int fieldSize = d->fields.size();
+  const int attributeSize = d->attributes.size();
+  if ( fieldSize != attributeSize )
+  {
+    QgsDebugMsg( QStringLiteral( "Attribute size (%1) does not match number of fields (%2)" ).arg( attributeSize ).arg( fieldSize ) );
+  }
+
+  for ( int i = 0; i < attributeSize && i < fieldSize; ++i )
```

Comment:
```suggestion
  if ( fieldSize != attributeSize )
  {
    QgsDebugMsg( QStringLiteral( "Attribute size (%1) does not match number of fields (%2)" ).arg( attributeSize ).arg( fieldSize ) );
    return QVariantMap();
  }

  for ( int i = 0; i < attributeSize; ++i )
```

That's what I meant to propose